### ISSUE TITLE
Traceback: pkg_resources.get_distribution("pywbem").version

### ISSUE THREAD

**Issue by storm49152** [Napsty/check_esxi_hardware#26]
Our system is still on CentOS-6 with Python-2.6 and pywbem-0.7.0, and we were having issues when connecting to a recent Dell OMSA. We updated this script to 20170905 and pywbem to 0.11.0 using pip. But after that the script fails:
```
# python ./check_esxi_hardware.py.orig --host=<fqdn> --user <user> --pass=<pass> -v dell
20180126 10:51:38 LCD Status: True
20180126 10:51:38 Connection to https://<fqdn>
Traceback (most recent call last):
  File "./check_esxi_hardware.py.orig", line 624, in <module>
    pywbemversion = pkg_resources.get_distribution("pywbem").version
  File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 292, in get_distribution
    if isinstance(dist,Requirement): dist = get_provider(dist)
  File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 176, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 648, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.6/site-packages/pkg_resources.py", line 546, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: pywbem
```

The problem is in line 624, where it says:
```
pywbemversion = pkg_resources.get_distribution("pywbem").version
```

I'm not well-versed in Python, but I could make the script work when I change this into
```
pywbemversion = pywbem.__version__
```

Diff:
```
--- check_esxi_hardware.py.orig 2017-09-05 21:17:13.000000000 +0200
+++ check_esxi_hardware.py      2018-01-26 10:50:45.416058045 +0100
@@ -621,7 +621,7 @@
 # connection to host
 verboseoutput("Connection to "+hosturl)
 # pywbem 0.7.0 handling is special, some patched 0.7.0 installations work differently
-pywbemversion = pkg_resources.get_distribution("pywbem").version
+pywbemversion = pywbem.__version__
 verboseoutput("Found pywbem version "+pywbemversion)
 if '0.7.' in pywbemversion:
   try:
```

I don't know if this is the preferred way of doing things, but at least it solves the issue for us.

**Comment by Napsty** [comment]
Thank you @storm49152 ! I wasn't aware of that python-internal function to show third party modules versions but found the relevant info here: https://www.python.org/dev/peps/pep-0396/

However there's a problem. The "original" pywbem 0.7 didn't include this version function:

```
admck@monitoring:~$ python
Python 2.7.6 (default, Oct 26 2016, 20:30:19) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pywbem
>>> pywbem.__version__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute '__version__'
```

This happened on a Ubuntu 14.04 LTS with the stock pywbem package. I generally like your idea and would prefer to use __version__. We could build a "try" clause around it though. 
First try to determine the pywbem version using the internal __version__ function. If it fails, use pkg_resources. 
What's your thought on that?

**Comment by Napsty** [comment]
Fixed in commit 277206abc770da9a5305082dfcb868a00f7d5388

**Comment by storm49152** [comment]
Hi. Well, it's like I said: I'm not very well-versed in Python and I had help finding the proposed solution. To me your solution looks good and I don't have any better ideas, so I'm fine with it. ;-)
Thanks for fixing it for the future!
